src/widgets/chart.rs

```diff
@@ -301,9 +301,15 @@ impl<'a> Chart<'a> {
             }
         }
 
-        if let Some(inner_width) = self.datasets.iter().map(|d| d.name.width() as u16).max() {
+        if let Some(inner_width) = self
+            .datasets
+            .iter()
+            .filter_map(|d| Some(d.name.as_ref()?.width() as u16))
+            .max()
+        {
             let legend_width = inner_width + 2;
-            let legend_height = self.datasets.len() as u16 + 2;
+            let legend_height =
+                self.datasets.iter().filter(|d| d.name.is_some()).count() as u16 + 2;
```

Comment:
There's some complexity to this that's probably worth calling to make it easier to understand for a future reader out by adding a comment / intermediate variable with a good name / extracted method with a good name.